### PR TITLE
ServiceBus Namespace restructured

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Restructured Service Bus Namespace to child-resource structure
- Updated ReadMe Generation Skript to better work with object references

Pipeline reference:
[ServiceBus: Namespaces](https://github.com/Azure/ResourceModules/actions/workflows/ms.servicebus.namespaces.yml)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

**Review by rahalan — APPROVED** [review]

File: arm/Microsoft.ServiceBus/namespaces/queues/deploy.bicep [hunk 1]
```diff
@@ -123,33 +123,39 @@ resource serviceBusNamespaceQueue 'Microsoft.ServiceBus/namespaces/queues@2021-0
...
     builtInRoleNames: builtInRoleNames
-    resourceName: serviceBusNamespaceQueue.name
+    resourceName: queue.name
   }
 }]
 
-output namespaceQueueName string = serviceBusNamespaceQueue.name
-output namespaceQueueResourceId string = serviceBusNamespaceQueue.id
+output namespaceQueueName string = queue.name
+output namespaceQueueResourceId string = queue.id
```
Comment (MariusStorhaug) [inline arm/Microsoft.ServiceBus/namespaces/queues/deploy.bicep:160]: Dont have to fix it here, but are we going to have outputs on all modules? Want to fix that here or a separate Issue/PR?

File: arm/Microsoft.ServiceBus/namespaces/virtualNetworkRules/readme.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+| Resource Type | Api Version |
+| :-- | :-- |
+| `Microsoft.ServiceBus/namespaces/virtualnetworkrules` | 2018-01-01-preview |
+
+## Parameters
+
+| Parameter Name | Type | Default Value | Possible Values | Description |
+| :-- | :-- | :-- | :-- | :-- |
+| `cuaId` | string |  |  | Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered |
+| `name` | string | `[format('{0}-vnr', parameters('namespaceName'))]` |  | Optional. The name of the virtual network rule |
```
Comment (MariusStorhaug) [inline arm/Microsoft.ServiceBus/namespaces/virtualNetworkRules/readme.md:16]: :D Interesting default value. Should we add an issue for improving the default value output in the script that generates the readme?

File: arm/Microsoft.ServiceBus/namespaces/authorizationRules/deploy.bicep [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  'Listen'
+  'Manage'
+  'Send'
+])
+param rights array = []
+
+@description('Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered')
+param cuaId string = ''
+
+module pid_cuaId './.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
```
Comment (MariusStorhaug) [inline arm/Microsoft.ServiceBus/namespaces/authorizationRules/deploy.bicep:20]: Remove `./`

File: arm/Microsoft.ServiceBus/namespaces/deploy.bicep [hunk 4]
```diff
@@ -244,5 +299,4 @@ module serviceBusNamespace_rbac '.bicep/nested_rbac.bicep' = [for (roleAssignmen
 output serviceBusNamespaceResourceId string = serviceBusNamespace.id
 output serviceBusNamespaceResourceGroup string = resourceGroup().name
 output serviceBusNamespaceName string = serviceBusNamespace.name
-output defaultAuthorizationRuleId string = defaultAuthorizationRuleId
 output serviceBusConnectionString string = 'Endpoint=sb://${serviceBusNamespaceName_var}.servicebus.windows.net/;SharedAccessKeyName=${listkeys(resourceId('Microsoft.ServiceBus/namespaces/authorizationRules', serviceBusNamespaceName_var, 'RootManageSharedAccessKey'), '2017-04-01').primaryKey}'
```
Comment (MariusStorhaug) [inline arm/Microsoft.ServiceBus/namespaces/deploy.bicep:302]: Missing output description. Choose to address here or open an issue for it

File: arm/Microsoft.ServiceBus/namespaces/disasterRecoveryConfigs/deploy.bicep [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+@description('Optional. Primary/Secondary eventhub namespace name, which is part of GEO DR pairing')
+param alternateName string = ''
+
+@description('Optional. ARM Id of the Primary/Secondary eventhub namespace name, which is part of GEO DR pairing')
+param partnerNamespace string = ''
+
+@description('Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered')
+param cuaId string = ''
+
+module pid_cuaId './.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
```
Comment (MariusStorhaug) [inline arm/Microsoft.ServiceBus/namespaces/disasterRecoveryConfigs/deploy.bicep:18]: Remove `./`

File: arm/Microsoft.ServiceBus/namespaces/ipFilterRules/deploy.bicep [hunk 6]
```diff
@@ -0,0 +1,46 @@
+@description('Required. Name of the parent Service Bus Namespace for the Service Bus Queue.')
+@minLength(6)
+@maxLength(50)
+param namespaceName string
+
+@description('Optional. The name of the ip filter rule')
+param name string = filterName
```
Comment (MariusStorhaug) [inline arm/Microsoft.ServiceBus/namespaces/ipFilterRules/deploy.bicep:7]: Something is strange here. Its depending on a required input?

File: arm/Microsoft.ServiceBus/namespaces/ipFilterRules/deploy.bicep [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+@description('Required. IP Filter name')
+param filterName string
+
+@description('Required. IP Mask')
+param ipMask string
+
+@description('Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered')
+param cuaId string = ''
+
+module pid_cuaId './.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
```
Comment (MariusStorhaug) [inline arm/Microsoft.ServiceBus/namespaces/ipFilterRules/deploy.bicep:25]: Remove `./`

File: arm/Microsoft.ServiceBus/namespaces/migrationConfigurations/deploy.bicep [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+@description('Required. Name to access Standard Namespace after migration')
+param postMigrationName string
+
+@description('Required. Existing premium Namespace ARM Id name which has no entities, will be used for migration')
+param targetNamespace string
+
+@description('Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered')
+param cuaId string = ''
+
+module pid_cuaId './.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
```
Comment (MariusStorhaug) [inline arm/Microsoft.ServiceBus/namespaces/migrationConfigurations/deploy.bicep:18]: Remove `./`

**Review by MariusStorhaug — CHANGES_REQUESTED** [review]
Some undeclared changes going on here, i.e cleanup naming of multiple resources in readme's ?
sxx-az-subnet-weu-x-001 -> sxx-az-subnet-x-001 ?

Over all changes:
- Remove `./`
- Add description to outputs
- Some stuff I have never seen on defaults on parameters
- Default value generated by readme script when there is a formula resulting in some crazy stuff in readme.

File: arm/Microsoft.ServiceBus/namespaces/queues/deploy.bicep (hunk 1 above)
Comment (MrMCake) [reply arm/Microsoft.ServiceBus/namespaces/queues/deploy.bicep]: fixed

File: arm/Microsoft.ServiceBus/namespaces/virtualNetworkRules/readme.md (hunk 2 above)
Comment (MrMCake) [reply arm/Microsoft.ServiceBus/namespaces/virtualNetworkRules/readme.md:16]: Yes, though I'm not sure right now how we could solve that. Check for [parameters]?

File: arm/Microsoft.ServiceBus/namespaces/authorizationRules/deploy.bicep (hunk 3 above)
Comment (MrMCake) [reply arm/Microsoft.ServiceBus/namespaces/authorizationRules/deploy.bicep]: done

File: arm/Microsoft.ServiceBus/namespaces/deploy.bicep (hunk 4 above)
Comment (MrMCake) [reply arm/Microsoft.ServiceBus/namespaces/deploy.bicep:309]: added

File: arm/Microsoft.ServiceBus/namespaces/disasterRecoveryConfigs/deploy.bicep (hunk 5 above)
Comment (MrMCake) [reply arm/Microsoft.ServiceBus/namespaces/disasterRecoveryConfigs/deploy.bicep]: fixed

File: arm/Microsoft.ServiceBus/namespaces/ipFilterRules/deploy.bicep (hunk 7 above)
Comment (MrMCake) [reply arm/Microsoft.ServiceBus/namespaces/ipFilterRules/deploy.bicep]: fixed

File: arm/Microsoft.ServiceBus/namespaces/ipFilterRules/deploy.bicep (hunk 6 above)
Comment (MrMCake) [reply arm/Microsoft.ServiceBus/namespaces/ipFilterRules/deploy.bicep:7]: Jap. The name is optional. The logic was that you can provide a dedicated name and if not the logic just re-uses the filtername

File: arm/Microsoft.ServiceBus/namespaces/migrationConfigurations/deploy.bicep (hunk 8 above)
Comment (MrMCake) [reply arm/Microsoft.ServiceBus/namespaces/migrationConfigurations/deploy.bicep]: done

**Review by MariusStorhaug — APPROVED** [review]
Comments resolved. Approved

**Pull request merged**